third_party/accessibility/base/string_utils.cc

```diff
@@ -18,6 +20,48 @@
 
 namespace base {
 
+using double_conversion::DoubleToStringConverter;
+using double_conversion::StringBuilder;
+
+namespace {
+constexpr char kExponentChar = 'e';
+constexpr char kInfinitySymbol[] = "Infinity";
+constexpr char kNaNSymbol[] = "NaN";
+
+// The number of digits after the decimal we allow before switching to
+// exponential representation.
+constexpr int kDecimalInShortestLow = -6;
+// The number of digits before the decimal we allow before switching to
+// exponential representation.
+constexpr int kDecimalInShortestHigh = 12;
+constexpr int kConversionFlags =
+    DoubleToStringConverter::EMIT_POSITIVE_EXPONENT_SIGN;
+
+const DoubleToStringConverter& GetDoubleToStringConverter() {
+  static DoubleToStringConverter converter(
+      kConversionFlags, kInfinitySymbol, kNaNSymbol, kExponentChar,
+      kDecimalInShortestLow, kDecimalInShortestHigh, 0, 0);
+  return converter;
+}
+
+std::string NumberToStringImpl(double number, bool is_single_precision) {
+  if (number == 0.0) {
+    return "0";
+  }
+
+  constexpr int kBufferSize = 128;
+  std::vector<char> char_buffer(kBufferSize);
```

Comment:
We should be able to use `std::array` here since we've got the size here as a constexpr; that'll compile down to the same code as a C style array, but with safer compile-time behaviour.